src/ui/public/state_management/state.js

```diff
@@ -230,23 +221,28 @@ export default function StateProvider(Private, $rootScope, $location, config) {
       return rison.encode(state);
     }
 
-    try {
-      const hash = this._hashingStore.hashAndSetItem(state);
+    // We need to strip out Angular-specific properties.
+    const json = angular.toJson(state);
+    const hash = createStateHash(json);
+    const isItemSet = hashedItemStore.setItem(hash, json);
```

Comment:
My original concern was the implicit dependency upon HashedItemStore's implementation (using sessionStorage). Any change there will necessitate a change here, but I can add a comment about that.